src/sio.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { RP2040, SIO_START_ADDRESS } from './rp2040';
+
+//HARDWARE DIVIDER
+const _DIV_UDIVIDEND = SIO_START_ADDRESS + 0x060; //  Divider unsigned dividend
+const _DIV_UDIVISOR = SIO_START_ADDRESS + 0x064; //  Divider unsigned divisor
+const _DIV_SDIVIDEND = SIO_START_ADDRESS + 0x068; //  Divider signed dividend
+const _DIV_SDIVISOR = SIO_START_ADDRESS + 0x06c; //  Divider signed divisor
+const _DIV_QUOTIENT = SIO_START_ADDRESS + 0x070; //  Divider result quotient
+const _DIV_REMAINDER = SIO_START_ADDRESS + 0x074; //Divider result remainder
+const _DIV_CSR = SIO_START_ADDRESS + 0x078;
```

Comment:
Can you please remove the _ (underscore) in front of the constant? The idea is to keep this code consistent with the rest of the code

Reply:
ok, I'm moving test to be gdbclient ready, I'll do it in the next commit.

Reply:
Oh great. I was going to suggest that, but you are faster!

Reply:
ok done, 
I had to add several "| 0"which are needed when running on silicone and the read value is signed, maybe a readSInt32 implementation could be better? or maybe not as it matters only during test? 

copied the silicone behavior when divisor=0 -> quot=-1, rem=dividend
I wasn't able to find any word related to div/0 exceptions in the datasheets, is this possible?

_DIV_XXX has been renamed to SIO_DIV_XXX, I choose this way as SIO_DIV_XXX = SIO_ADDRESS_BASE + DIV_XXX
Even if declared in different files, I wouldn't have the same name to both offset and full address.

Reply:
Thank you!

> copied the silicone behavior when divisor=0 -> quot=-1, rem=dividend
Good idea to test what happens when dividing by zero. 

> I wasn't able to find any word related to div/0 exceptions in the datasheets, is this possible?

No idea! Did you see it getting any exception on the silicone? 

> I had to add several "| 0"which are needed when running on silicone and the read value is signed, maybe a readSInt32 implementation could be better? or maybe not as it matters only during test?

I think this is a good idea, it will make the tests easier to read / understand. I suggest calling it `readInt32()` (without the S), similar to how they call the [signed int array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int32Array) in JavaScript (and also C++ type [int32_t vs uint32_t](https://en.cppreference.com/w/cpp/types/integer)).

> _DIV_XXX has been renamed to SIO_DIV_XXX, I choose this way as SIO_DIV_XXX = SIO_ADDRESS_BASE + DIV_XXX
Even if declared in different files, I wouldn't have the same name to both offset and full address.

👍